### PR TITLE
Arrays - lucky number

### PR DESCRIPTION
**Pull request by DevratAgarwal**
## Checklist
- Solution is tested on sphere engine <br/>
    `Note:` Please copy only the solution part. Do not copy/paste the template code. That way, any issues with template code can be caught early on.
    - [ ] Author
    - [ ] Reviewer
- PR title is correct - [PR title conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#title-for-a-pull-request)
    - [ ] Author
    - [ ] Reviewer
- Folder/File naming is correct - [Naming conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#naming-of-files-and-folders)
    - [ ] Author
    - [ ] Reviewer
- Coding conventions are followed - [Coding conventions](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#coding-conventions)
    - [ ] Author
    - [ ] Reviewer
- Test cases (action item for **reviewer**) <br/>
    - To be ensured by checking the problem on sphere-engine
    - If any test cases are missing, please add them - [how to add new test cases](https://github.com/DSA-solutions-the10xacademy/JavaScript-DSA-Solutions#adding-new-test-cases)
    - [ ] There are at least 2 test cases
    - [ ] There is at least 1 test case that is different from the sample test cases

### CODE REVIEW

**Review by LVSsandeepVarma — COMMENTED** [review]
@DevratAgarwal  I have updated the template code  , please  change it accordingly

File: solutions/arrays/lucky-number [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+// 			  count=count+1
+// 		  }
+// 	  }
+// 	  if(count===nums[index])
+// 	  {
+// 		  return count
+// 	  }
+//   }
+//   return -1
+
```
Comment (LVSsandeepVarma) [inline solutions/arrays/lucky-number:36]: @DevratAgarwal  since the array is already sorted, we can do it with linear time complexity,

Comment (DevratAgarwal) [reply solutions/arrays/lucky-number:36]: @LVSsandeepVarma done

Comment (LVSsandeepVarma) [reply solutions/arrays/lucky-number:36]: @DevratAgarwal  I have updated the template code, please update it and we can do this problem in O(n) time  complexity

Comment (DevratAgarwal) [reply solutions/arrays/lucky-number:36]: @LVSsandeepVarma code is in O(n) time complexity

**Comment by sailesh-ramayanam** [comment]
@LVSsandeepVarma Please check how many test cases fail for a brute force solution (i.e. `O(n^2)`).

File: solutions/arrays/lucky-number [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	let lastElement = -1, numberOfOccurences = 0;
+	for (let index = 0; index < nums.length; index++) {
+		if (nums[index] === lastElement) {
+			numberOfOccurences++;
+		}
+		else {
+			lastElement = nums[index];
+			numberOfOccurences = 1;
+		}
+		if (numberOfOccurences === lastElement)
```
Comment (sailesh-ramayanam) [inline solutions/arrays/lucky-number:25]: @DevratAgarwal This solution will fail for `1 1 2`. Expected output is `-1`, where as this solution will give `1`.

**Review by sailesh-ramayanam — CHANGES_REQUESTED** [review]
@DevratAgarwal This solution will fail for `1 1 2`. Expected output is `-1`, where as this solution will give `1`.

**Comment by DevratAgarwal** [comment]
@sailesh-ramayanam it  is giving -1.

**Comment by sailesh-ramayanam** [comment]
> @sailesh-ramayanam it is giving -1.

image

**Comment by LVSsandeepVarma** [comment]
> @LVSsandeepVarma Please check how many test cases fail for a brute force solution (i.e. `O(n^2)`).

@sailesh-ramayanam  only one testcase is getting TLE for solution with time complexity of  O(n^2)

**Review by LVSsandeepVarma — APPROVED** [review]

**Pull request closed without merging**